dashboard/controller.py

```diff
@@ -88,6 +97,29 @@ def show_builds_history(page=1):
                            buildsets_log_url=buildset_log_url)
 
 
+@build_enqueue.route('/control')
+def control_panel():
+    if session and session['user'] and session['user'].is_admin():
+        return render_template('control_panel.html')
+    raise requests.codes.forbidden
+
+
+@build_enqueue.route('/start/<pipeline>/<branch>')
+def start_buildset(pipeline, branch):
```

Comment:
IMO this particular route should be managed by `POST` request, `not GET`.
Because you are sending specific information to backend with request to do external action on these information.